### PR TITLE
Feature/moretests

### PR DESCRIPTION
**Pull request by enola-dkfz**

### CODE REVIEW

File: src/main/java/de/samply/transfair/converters/TemperatureConverter.java [hunk 1]
```diff
@@ -38,22 +38,26 @@ public static Extension fromMiiToBbmri(Long high, Long low) {
     Extension extension = new Extension();
     extension.setUrl("https://fhir.bbmri.de/StructureDefinition/StorageTemperature");
 
+    CodeableConcept codeableConcept = new CodeableConcept();
+
     if (high <= 10 && low >= 2) {
-      extension.setValue(new CodeableConcept().getCodingFirstRep().setCode("temperature2to10"));
+      codeableConcept.getCodingFirstRep().setSystem("https://fhir.bbmri.de/CodeSystem/StorageTemperature").setCode("temperature2to10");
```
Comment (patrickskowronekdkfz) [inline src/main/java/de/samply/transfair/converters/TemperatureConverter.java:44]: Please use a constant for the URL

File: src/main/java/de/samply/transfair/resources/SpecimenMapping.java [hunk 2]
```diff
@@ -143,6 +143,19 @@ public void fromMii(org.hl7.fhir.r4.model.Specimen resource) {
         this.setMiiConditionRef(extension.getValue().toString());
       }
     }
+
+
+    // Storage temperature is an extension of the collection
```
Comment (patrickskowronekdkfz) [inline src/main/java/de/samply/transfair/resources/SpecimenMapping.java:148]: The collection doesn't have an extension temperutre. It has blood fusions extensions, which isn't a field on the bbmri.de side.

File: src/test/java/de/samply/transfair/JsonUtils.java [hunk 3]
```diff
@@ -13,6 +13,8 @@ public class JsonUtils {
   public static void compareFhirObjects(IBaseResource a, IBaseResource b) {
     String actualAsJson = parser.encodeResourceToString(a);
     String expectedAsJson = parser.encodeResourceToString(b);
+    log.info(actualAsJson);
```
Comment (patrickskowronekdkfz) [inline src/test/java/de/samply/transfair/JsonUtils.java:16]: Please don't log the json.

File: src/test/java/de/samply/transfair/resources/SpecimenMappingTest.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    storageTemperatureExtensionBbmri.setValue(codeableConceptBbmriForConvertingStorageTemperature);
+    specimenBbmriForConverting.setExtension(List.of(diagnosisExtensionBbmri, storageTemperatureExtensionBbmri));
+
+    CodeableConcept codeableConceptBbmriForConvertingSampleType = new CodeableConcept();
+    codeableConceptBbmriForConvertingSampleType.getCodingFirstRep().setSystem("https://fhir.bbmri.de/CodeSystem/SampleMaterialType").setCode("blood-plasma");
+    specimenBbmriForConverting.setType(codeableConceptBbmriForConvertingSampleType);
+    specimenBbmriForConverting.getCollection().getCollectedDateTimeType().setValue(collectedDate);
+
+
+    specimenMiiForComparing = new Specimen();
```
Comment (patrickskowronekdkfz) [inline src/test/java/de/samply/transfair/resources/SpecimenMappingTest.java:79]: fastingStatus is missing.

File: src/test/java/de/samply/transfair/resources/SpecimenMappingTest.java [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    cal.set(Calendar.HOUR, 0);
+    cal.set(Calendar.MINUTE, 0);
+    cal.set(Calendar.SECOND, 0);
+    cal.set(Calendar.ZONE_OFFSET, 2);
+    Date collectedDate = cal.getTime();
+
+    //    DateTimeType collected = new DateTimeType();
+    //    collected.setValueAsString("1997-10-22T00:00:00+02:00");
+
+    specimenBbmriForConverting = new Specimen();
```
Comment (patrickskowronekdkfz) [inline src/test/java/de/samply/transfair/resources/SpecimenMappingTest.java:54]: fastingStatus is missing.

**Review by patrickskowronekdkfz — CHANGES_REQUESTED** [review]
Needs some small improvements, but overall it is looking good.

File: src/test/java/de/samply/transfair/JsonUtils.java (hunk 3 above)
Comment (enola-dkfz) [reply src/test/java/de/samply/transfair/JsonUtils.java]: done

File: src/test/java/de/samply/transfair/resources/SpecimenMappingTest.java (hunk 4 above)
Comment (enola-dkfz) [reply src/test/java/de/samply/transfair/resources/SpecimenMappingTest.java:83]: done

File: src/test/java/de/samply/transfair/resources/SpecimenMappingTest.java (hunk 5 above)
Comment (enola-dkfz) [reply src/test/java/de/samply/transfair/resources/SpecimenMappingTest.java:56]: done

File: src/test/java/de/samply/transfair/converters/TemperatureConverterTest.java [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+import org.hl7.fhir.r4.model.CodeableConcept;
+import org.hl7.fhir.r4.model.Extension;
+import org.hl7.fhir.r4.model.Quantity;
+import org.hl7.fhir.r4.model.Range;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;;
```
Comment (patrickskowronekdkfz) [inline src/test/java/de/samply/transfair/converters/TemperatureConverterTest.java:12]: One ";" too much

**Review by patrickskowronekdkfz — APPROVED** [review]

**Pull request merged**